### DISCUSSION TITLE
Possible limit for maximum characters? (old audio devices)

### DISCUSSION THREAD

**Discussion by maciulex** [shaked6540/YoutubePlaylistDownloader#264] [Ideas]
Context
In some old audio devices there is limit to card capacity and title lenght
in my particular case title can't be longer then 64 chars including .mp3
would it be possible to add some formating rule to trim $title?
i know it is pretty unique but it bothers to think that one song can be missed out

i created bat script with chatGPT to make quick fix.
if you need to change lenght you need to change 55 in **two diffrent places** :) (i know im lazy _shhh_ lets just overlook this)

in the end
it saved 23 song! well from collection of 250

```powershell
@echo off
setlocal enabledelayedexpansion


:: checking if path was given as argument
if "%~1"=="" (
    echo Please give path as argument
    exit /b
)

:: Saving path to variable
set "folder=%~1"

:: checking if path exists
if not exist "%folder%\" (
    echo given path do not exists
    exit /b
) ELSE (
    echo "test2"
)

:: going by every file in path
echo Lenght of filenames if folder: %folder%
for %%f in ("%folder%\*") do (
    set "fullfilename=%%~nxf"
    set "filename=%%~nf"
    set "extension=%%~xf"
    set "length=0"
    for /l %%i in (1,1,255) do (
        if defined filename (
            set "filename=!filename:~1!"
            set /a "length+=1"
        )
    )

    :: Checking file lenght and reducing it
    set "newfilename=!fullfilename!"
    if !length! gtr 55 (
        set "newfilename=!fullfilename:~0,55!!extension!"
        ren "%%f" "!newfilename!"
	  echo old name: !fullfilename! - new name: !newfilename! - name lenght: !length! znakow
    )
    
   
)

endlocal
```

**Comment by shaked6540** [comment]
Do you mean the file name can't be more than 64 characters? I can't make that change for everyone, because some videos will have titles longer than 64 characters.

In your case it seems like the script will be enough, sadly I don't have time to add new features anymore

**Comment by shaked6540** [comment]
Maybe I can add $titleshort or something that will be the first 60 characters of the title, and along with the extension it will be 64, I'll see if its possible to add it when I have time

**Comment by maciulex** [comment]
if you really dont have time then don't get bothered by it
i guess it is realllly small .% of people that can have this problem
and if they are determined enough they will try .bat script and get good enough outcome
